packages/block-library/src/heading/edit.js

```diff
@@ -72,3 +81,17 @@ export default function HeadingEdit( {
 		</Fragment>
 	);
 }
+
+export default withSelect( ( select ) => {
+	// Parse the h1,h2,h3... choices to level numbers and pass it as a prop.
+	const levelChoices = get(
+		select( 'core/blocks' ).getBlockType( 'core/heading' ),
+		[ 'attributes', 'level', 'enum' ],
+		[]
+	);
+
+	return {
+		levelChoices,
```

Comment:
This implementation will result in poor performance, at least when using the fallback empty array value, due to a combination of factors that (a) `withSelect` is called for any change in application state, (b) you are producing a new array reference value (`[] !== []`), and (c) a render is forced for any new prop value reference.

As long as a block types `attributes.level.enum` remains a consistent reference (I think it's safe to assume), then a simple adjustment here may be to move the array default to the top of the file as a shared constant:

```js
const DEFAULT_LEVEL_CHOICES = [ 1, 2, 3, 4, 5, 6, 7 ];
```

Reply:
Thanks! Great point. It's fixed now.

Just a side comment: The previous `maxLevel` was set to 7 but that's confusing, `HeadingToolbar` component makes use of Lodash `range` https://lodash.com/docs/4.17.11#range and when it generates the range, the second argument passed to the function (7) is not included in the range. That's why we only need 1-6 as default values.